js/tji-datasets-explore.js

```diff
@@ -307,64 +309,124 @@ TJIChartView.prototype.transform_data = function() {
     if (values.indexOf(that.missing_data_label) !== -1) {
       values.splice(values.indexOf(that.missing_data_label), 1);
       values.push(that.missing_data_label);
     }
-    filters.push({
-      key: column,
-      values: values
-    });
+    filter_config.values = values;
   });
 
-  // Build out checkboxes n stuff
+  // Build out filters
   var fieldsets = [];
-  _.each(filters, function(f) {
-    var fieldset = jQuery('<fieldset><legend>' + f.key.replace(/_/g, " ") + '<i class="fas fa-caret-down"></i></legend></fieldset>');
-    _.each(f.values, function(v) {
-      var input = jQuery('<input/>', {
-        type: "checkbox",
-        checked: "checked",
-        id: v,
-        name: f.key,
-        value: v,
-      });
-      var label = jQuery('<label/>', {
-        for: v,
-      }).text(v);
-      fieldset.append(jQuery('<div></div>').append(input, label));
-    });
+  _.each(that.filter_configs, function(filter_config) {
+    var fieldset;
+    switch(filter_config.type) {
+      case 'autocomplete': 
+        fieldset = that.create_filter_autocomplete(filter_config);
+        break;
+      case 'checkbox':
+      default:
+        fieldset = that.create_filter_checkboxes(filter_config);
+        break;
+    }
     fieldsets.push(fieldset);
   });
   
-  this.$form = jQuery('<form />').append(fieldsets);
+  this.$form = jQuery('<form />', {
+    class: 'tji-chart-filters',
+  }).append(fieldsets);
   this.state.active_filters = this.$form.serializeArray();  
-  jQuery(this.filters_elt_id).append(this.$form);
+  jQuery(this.filters_elt_selector).append(this.$form);
   
 }
 
+TJIChartView.prototype.create_filter_checkboxes = function(filter) {
+  var that = this;
+  var fieldset = jQuery('<fieldset><legend>' + filter.name.replace(/_/g, " ") + '<i class="fas fa-caret-down"></i></legend></fieldset>');
+  _.each(filter.values, function(v) {
+    fieldset.append(that.create_filter_checkbox(filter.name, v));
+  });
+  return fieldset;
+}
+
+TJIChartView.prototype.create_filter_checkbox = function(name, value) {
+  var input = jQuery('<input/>', {
+      class: "tji-chart-filters__checkbox",
+      type: "checkbox",
+      checked: "checked",
+      id: 'TJIChartView__filter-' + value,
+      name: name,
+      value: value,
+    });
+  var label = jQuery('<label/>', {
+    for: 'TJIChartView__filter-' + value,
+  }).text(value);
+  var container = jQuery('<div/>', {
+      class: "tji-chart-filters__filter",
+  }).append(input, label);
+  return container;
+}
+
+TJIChartView.prototype.create_filter_autocomplete = function(filter) {
+  var that = this;
+  var fieldset = jQuery('<fieldset><legend>' + filter.name.replace(/_/g, " ") + '<i class="fas fa-caret-down"></i></legend></fieldset>');
+  var input = jQuery('<input/>', {
+    class: "tji-chart-filters__text",
+    type: "text",
+  });
+  fieldset.append(jQuery('<div/>',{
+    class: "tji-chart-filters__autocomplete",
+  }).append(input));
+  var $auto_complete = jQuery('<div />', {
```

Comment:
Minor: I think there's risk for confusion using the exact same variable name (except for the `$`) -- could you call this `auto_complete_jquery` and the next one `auto_complete_widget`, or something?

Reply:
I can, but this notation is pretty common in the javascript world.

Reply:
![you](https://user-images.githubusercontent.com/1421476/42423858-3b7b7b94-82c7-11e8-9d83-ae3e59c3195f.gif)